ta2resources/client_creator.py

```diff
@@ -125,7 +127,7 @@ def store_record(self, query: Search, record: Record, conn):
         if not exists or not local_s3:
             temp_file = "{}-temp".format(record_id)
             self.client.read_file(record_id, temp_file)
-            object_location = "{}-s3".format(record_id)
+            object_location = "s3-{0}-{1}".format(record_id, file_name)
```

Comment:
and then 
`object_location = "s3-{id}{hyphen}{name}".format(id=record_id, hyphen="-" if file_name else "", name=file_name)`

This means that we aren't presupposing all unnamed files are text files